crates/astria-sequencer/src/app.rs

```diff
@@ -426,22 +429,206 @@ impl App {
         (signed_txs, validated_txs)
     }
 
+    #[instrument(name = "App::finalize_block", skip_all)]
+    pub(crate) async fn finalize_block(
+        &mut self,
+        finalize_block: abci::request::FinalizeBlock,
+        storage: Storage,
+    ) -> anyhow::Result<abci::response::FinalizeBlock> {
+        use tendermint::{
+            abci::types::ExecTxResult,
+            block::Header,
+        };
+
+        use crate::transaction::InvalidNonce;
+
+        let data_hash = astria_core::sequencerblock::v1alpha1::block::merkle_tree_from_data(
+            finalize_block.txs.iter().map(std::convert::AsRef::as_ref),
+        )
+        .root();
+
+        let chain_id: tendermint::chain::Id = self
+            .state
+            .get_chain_id()
+            .await
+            .context("failed to get chain ID from state")?
+            .try_into()
+            .context("invalid chain ID")?;
+
+        // call begin_block on all components
+        // NOTE: the fields marked `unused` are not used by any of the components;
+        // however, we need to still construct a `BeginBlock` type for now as
+        // the penumbra IBC implementation still requires it as a parameter.
+        let begin_block = abci::request::BeginBlock {
+            hash: finalize_block.hash,
+            byzantine_validators: finalize_block.misbehavior.clone(),
+            header: Header {
+                app_hash: self.app_hash.clone(),
+                chain_id: chain_id.clone(),
+                consensus_hash: Hash::default(), // unused
+                data_hash: Some(Hash::try_from(data_hash.to_vec()).unwrap()),
+                evidence_hash: Some(Hash::default()), // unused
+                height: finalize_block.height,
+                last_block_id: None,                      // unused
+                last_commit_hash: Some(Hash::default()),  // unused
```

Comment:
Turns out, this is just: `Some(Hash::default()) -> Some(Hash::None)`

Reply:
yes :)